internal/dataplane/parser/translators/ingress.go

```diff
@@ -289,3 +292,36 @@ func flattenMultipleSlashes(path string) string {
 	}
 	return string(out)
 }
+
+// legacyRegexPathExpression is the regular expression used by Kong <3.0 to determine if a path is not a regex.
+var legacyRegexPathExpression = regexp.MustCompile(`^[a-zA-Z0-9\.\-_~/%]*$`)
+
+// MaybePrependRegexPrefix takes a path, controller regex prefix, and a legacy heuristic toggle. It returns the path
+// with the Kong regex path prefix if it either began with the controller prefix or did not, but matched the legacy
+// heuristic, and the heuristic was enabled.
+func MaybePrependRegexPrefix(path, controllerPrefix string, applyLegacyHeuristic bool) string {
```

Comment:
Given that this and `MaybePrependRegexPrefixForIngressV1Fn()` below are well isolated,  exported `func`s can we add UTs for them as well? The logic they implement seems neither the most straightforward nor the most convoluted but I believe a handful of UTs wouldn't hurt but on the contrary, help reader understand what can be expected from particular set of inputs to those.

Reply:
Good point :)

Reply:
PTAL 9b48c9c